docs/source/poet/sgx_primer.rst

```diff
@@ -0,0 +1,208 @@
+
+Within SGX, attestation is, “the mechanism by which another party can gain
+confidence that the correct software is running within an enclave on an enabled
+platform.” [2]_
+
+Enclave Reports
+---------------
+
+Enclaves can generate a data structure referred to as a "report" which can be
+used for attestation. When creating a report, the measurement of the target
+enclave (the intended receiver of the report) is required. Enclave reports
+contain:
+
+* The enclave’s identity (Sealing Authority and Enclave Measurement)
+* Additional enclave attributes
+* Additional developer-defined information
+* A message authentication code (MAC)
+
+Intra-Platform Attestation
+--------------------------
+
+Intra-Platform attestation refers to attestation between two enclaves on the
+same platform. That is, they are both running on the same Intel® SGX enabled
+processor and host.
+
+1. A report is created for a given target enclave on the same platform.
+2. The target enclave verifies the report.
+3. A secure channel is then constructed between the two enclaves using standard
+   encryption methods.
+
+Inter-Platform Attestation
+--------------------------
+
+Inter-Platform attestation refers to attestation between two enclaves running on
+different platforms, most likely connected by a network connection. This form of
+attestation is used by PoET and Intel® provides an implementation of this service
+through the Intel® Attestation Service (IAS). It is also known as “remote
+attestation”.
+
+A special “Quoting Enclave” is provided with SGX and is used in Inter-process
+Attestation. The quoting enclave processes reports using the following
+procedure:
+
+* The Quoting Enclave verifies reports submitted to it from other enclaves on
+  the same platform using the Intra-Platform Attestation described above.
+* It then replaces the MAC with a signature created with an anonymous credential
+  tied to the device hardware.
+* This new data structure is called a “quote”.
+
+The Quoting Enclave uses the Intel® Enhanced Privacy ID (EPID) signature scheme
+to overcome the privacy concerns associated with using a small number of
+asymmetric keys for signing over the life of a platform. “EPID is a group
+signature scheme that allows a platform to sign objects without uniquely
+identifying the platform or linking different signatures.” [2]_ In “anonymous”
+mode, a signature verifier cannot associate a given signature with a given
+identity. In “pseudonymous” mode, a signature verifier can determine whether it
+has verified a platform previously. PoET uses the "pseudonymous" or "named base"
+mode. This provides a consistent identity for a given blockchain network, but
+but does not otherwise link an identity to other blockchain networks or other
+SGX applications.
+
+The remote attestation procedure is roughly the following:
+
+1. A local application enclave wants to interact with a remote enclave so it
+   connects to the remote service through the local application.
```

Comment:
The local application connects to itself? Perhaps reword this sentence for clarity?